lib/galaxy/datatypes/msa.py

```diff
@@ -11,6 +11,87 @@
 log = logging.getLogger(__name__)
 
 
+class Infernal_CM_1_1(Text):
+    file_ext = "cm"
+
+    MetadataElement(name="number_of_models", default=0, desc="Number of covariance models",
+                    readonly=True, visible=True, optional=True, no_value=0)
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = get_file_peek(dataset.file_name, is_multi_byte=is_multi_byte)
+            if dataset.metadata.number_of_models == 1:
```

Comment:
Wouldn't line17 be needed for setting the metadata here? Although idk if set_peek is overall needed or not

Reply:
Yes you can remove the model count completely, if we do not count anymore.

Reply:
I guess, we count number of models in line 204. It's a useful info.